azext_iot/iothub/providers/pnp_runtime.py

```diff
@@ -0,0 +1,101 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_iot.common.shared import SdkType
+from azext_iot.common.utility import unpack_msrest_error, process_json_arg
+from azext_iot.iothub.providers.base import (
+    IoTHubProvider,
+    CloudError,
+)
+
+
+logger = get_logger(__name__)
+
+
+class PnPRuntimeProvider(IoTHubProvider):
+    def __init__(self, cmd, hub_name=None, rg=None, login=None):
+        super(PnPRuntimeProvider, self).__init__(
+            cmd=cmd, hub_name=hub_name, rg=rg, login=login
+        )
+        self.runtime_sdk = self.get_sdk(SdkType.pnp_sdk).digital_twin
+
+    def invoke_device_command(
+        self,
+        device_id,
+        command_name,
+        timeout=30,
+        payload="{}",
+        component_path=None,
+        connect_timeout=None,
+        response_timeout=None,
+    ):
+
+        # Prevent msrest locking up shell
+        self.runtime_sdk.config.retry_policy.retries = 1
+
+        try:
+            if payload:
+                payload = process_json_arg(payload, argument_name="payload")
+
+            api_timeout_kwargs = {
```

Comment:
might be worth adding `absoloute_timeout_in_seconds` i.e. "stop listening after 10 seconds regardless of whats going on"

Reply:
Will look into something like that in a future iteration.